client/my-sites/plans-features-main/hooks/data-store/use-pricing-meta-for-grid-plans.ts

```diff
@@ -119,14 +117,15 @@ const usePricingMetaForGridPlans: UsePricingMetaForGridPlans = ( {
 					 */
 					if ( purchasedPlan ) {
 						const isMonthly = purchasedPlan.billPeriodDays === PLAN_MONTHLY_PERIOD;
-						const isYearly = purchasedPlan.billPeriodDays === PLAN_ANNUAL_PERIOD;
 
 						if ( isMonthly && monthlyPrice !== purchasedPlan.priceInteger ) {
 							monthlyPrice = purchasedPlan.priceInteger;
-							yearlyPrice = calculateYearlyPrice( TERM_ANNUALLY, purchasedPlan.priceInteger );
-						} else if ( isYearly && yearlyPrice !== purchasedPlan.priceInteger ) {
-							monthlyPrice = calculateMonthlyPrice( TERM_ANNUALLY, purchasedPlan.priceInteger );
-							yearlyPrice = purchasedPlan.priceInteger;
+							fullPrice = parseFloat( ( purchasedPlan.priceInteger * 12 ).toFixed( 2 ) );
+						} else if ( fullPrice !== purchasedPlan.priceInteger ) {
+							const term = getTermFromDuration( purchasedPlan.billPeriodDays ) || '';
+							const months = getBillingMonthsForTerm( term );
+							monthlyPrice = parseFloat( ( purchasedPlan.priceInteger / months ).toFixed( 2 ) );
```

Comment:
Now that you have the term, the part above could be simplified by going back to using `calculateMonthlyPrice()` again?

Reply:
Sounds good. Added `calculateMonthlyPrice()` back to the mix.